### PR TITLE
Make README more useful

### PR DESCRIPTION
**Pull request by rajkan01**
* Provide description on application functionality
* Add official example badge
* Remove outdated information
* Link to Mbed boards
* Indicate how to find out which toolchain support a target chosen
* Provide instructions to flash after building
* Updated the tests README to give proper information with example

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,40 +1,133 @@
...
+```
+Num:            Number of times socket state changes and repeated the same value to indicate TCP and UDP
+ID:             Socket id
+State:          Socket states
+Protocol:       TCP, UDP
+Sent bytes:     Number of bytes sent through this socket  
+Received bytes: Number of bytes received through this socket
+Time in us:     When state last changed
+```
+## Network interface and Socket statistics target configuration
```
Comment (evedon) [inline README.md]: The content of this section should move under `Application functionality`

File: README.md [hunk 2]
```diff
@@ -1,40 +1,133 @@
-# Getting started with network socket statistics on Mbed OS
+![](./resources/official_armmbed_example_badge.png)
+# Socket Statistics Mbed OS Example
 
-This example demonstrates how you can collect statistics from network sockets. You can enable statistics with the `nsapi.socket-stats-enabled` configuration option in `mbed_app.json`:
+This guide reviews the steps required to get socket statistics on an Mbed OS enabled platform.
```
Comment (evedon) [inline README.md]:
```suggestion
This example demonstrates how you can collect statistics from network sockets on an Mbed OS enabled platform.
```

File: README.md [hunk 3]
```diff
@@ -1,40 +1,133 @@
...
+    ```
+
+    Alternatively, you can download the example project with Arm Mbed CLI using the `import` subcommand:
+
+    ```bash
+    $ mbed import mbed-os-example-socket-stats && cd mbed-os-example-socket-stats
+    ```
+
+## Application functionality
+
```
Comment (evedon) [inline README.md:25]: You can enable socket statistics by setting `nsapi.socket-stats-enabled` to true in `mbed_app.json`: _give example_

File: README.md [hunk 4]
```diff
@@ -1,40 +1,133 @@
...
+The `main()` function gets the default instance of a network interface and set up the socket connection. It also starts a thread to print socket statistics such as  `reference_id`, `state`, `proto`, `sent_bytes`, `recv_bytes`, `last_change_tick` on the serial interface.
+
```
Comment (evedon) [inline README.md:43]: The current example is configured to use the Ethernet interface on supported devices. To use the example with a different interface, please follow the [socket example readme](https://github.com/ARMmbed/mbed-os-example-sockets/blob/master/README.md).

File: README.md [hunk 5]
```diff
@@ -1,40 +1,133 @@
...
 
- 
-### Note
+## Related links
 
-The current example is configured to use the Ethernet interface on supported devices. To use the example with a different interface, please follow the [socket example readme](https://github.com/ARMmbed/mbed-os-example-sockets/blob/master/README.md). 
-Network interfaces are documented in [Mbed OS 5 documentation](https://os.mbed.com/docs/latest/apis/network-interfaces.html)
+* [Mbed OS network socket API](https://docs.mbed.com/docs/mbed-os-api-reference/en/latest/APIs/communication/network_sockets/).
+* [Mbed OS socket stats API](https://os.mbed.com/docs/latest/apis/socketstats.html).
+* [Mbed OS configuration](https://os.mbed.com/docs/latest/reference/configuration.html).
```
Comment (evedon) [inline README.md]: Also add:   [Network interfaces](https://os.mbed.com/docs/latest/apis/network-interfaces.html)

File: README.md [hunk 6]
```diff
@@ -1,40 +1,133 @@
...
+
+    Alternatively, you can download the example project with Arm Mbed CLI using the `import` subcommand:
+
+    ```bash
+    $ mbed import mbed-os-example-socket-stats && cd mbed-os-example-socket-stats
+    ```
+
+## Application functionality
+
+The `main()` function gets the default instance of a network interface and set up the socket connection. It also starts a thread to print socket statistics such as  `reference_id`, `state`, `proto`, `sent_bytes`, `recv_bytes`, `last_change_tick` on the serial interface.
```
Comment (evedon) [inline README.md]:
```suggestion
The `main()` function opens a socket on the Ethernet interface and creates a TCP connection to generate some data traffic. It also starts a thread to periodically print socket statistics such as  `reference_id`, `state`, `proto`, `sent_bytes`, `recv_bytes`, `last_change_tick` on the console.
```

File: main.cpp [hunk 7]
```diff
@@ -37,39 +37,40 @@ void print_stats()
         count = SocketStats::mbed_stats_socket_get_each(&stats[0], MBED_CONF_NSAPI_SOCKET_STATS_MAX_COUNT);
         for (int i = 0; i < count; i++) {
             stdio_mutex.lock();
-            printf("Num: %d", num);
-            printf(" ID: %p", stats[i].reference_id);
+            printf("Num: %d   ", num);
```
Comment (evedon) [inline main.cpp]: I would suggest to rename Num to `Iteration`

File: main.cpp [hunk 8]
```diff
@@ -169,16 +164,9 @@ int main()
         stdio_mutex.unlock();
         goto DISCONNECT;
     }
-    // the HTTP response code
-    stdio_mutex.lock();
-    printf("recv %d [%.*s]\n", rcount, strstr(buffer, "\r\n") - buffer, buffer);
-    stdio_mutex.unlock();
 
     // The api.ipify.org service also gives us the device's external IP address
     p = strstr(buffer, "\r\n\r\n") + 4;
```
Comment (evedon) [inline main.cpp]: This is no longer needed if you are not going to print the external IP address.

File: main.cpp [hunk 9]
```diff
@@ -169,16 +164,9 @@ int main()
...
-    stdio_mutex.lock();
-    printf("External IP address: %.*s\n", rcount - (p - buffer), p);
-    stdio_mutex.unlock();
 
 DISCONNECT:
     delete[] buffer;
```
Comment (evedon) [inline main.cpp]: I think that `buffer` can be removed if you take my earlier comment.

**Review by evedon — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -12,29 +35,102 @@ This example demonstrates how you can collect statistics from network sockets. Y
 }
 ```
 
-### Building
+The `main()` function opens a socket on the Ethernet interface and creates a TCP connection to generate some data traffic. It also starts a thread to periodically print socket statistics such as  `reference_id`, `state`, `proto`, `sent_bytes`, `recv_bytes`, `last_change_tick` on the serial interface.
```
Comment (evedon) [inline README.md]: You should use the variable names shown in the console, so `ID`, `State` etc.

File: README.md [hunk 11]
```diff
@@ -12,29 +35,102 @@ This example demonstrates how you can collect statistics from network sockets. Y
...
+Iteration: 13    ID: 0x2000e050    State: Closed    Proto: UDP    Sent: 31    Recv: 257    Time: 7615
+Iteration: 14    ID: 0x2000e208    State: Connected Proto: TCP    Sent: 58    Recv: 0    Time: 7815
+Iteration: 14    ID: 0x2000e050    State: Closed    Proto: UDP    Sent: 31    Recv: 257    Time: 7615
+Iteration: 15    ID: 0x2000e208    State: Connected Proto: TCP    Sent: 58    Recv: 0    Time: 7815
+Iteration: 15    ID: 0x2000e050    State: Closed    Proto: UDP    Sent: 31    Recv: 257    Time: 7615
+```
+The information below shows how to interpret the above fields:
+
+```
+Iteration:      Number of times socket state changes and repeated the same value to indicate TCP and UDP
```
Comment (evedon) [inline README.md]:
```suggestion
Iteration:      Number of times socket state changes
```

**Review by evedon — APPROVED** [review]

**Review by AnotherButler — APPROVED** [review]
Thanks for the PR 👍

**Pull request merged**